test/e2e/kuma_deploy_hybrid_kube_global_test.go

```diff
@@ -77,7 +80,7 @@ var _ = Describe("Test Kubernetes/Universal deployment when Global is on K8S", f
 
 	It("communication in between apps in remote zone works", func() {
 		stdout, _, err := remoteCluster.ExecWithRetries("", "", "demo-client",
-			"curl", "-v", "-m", "3", "localhost:4001")
+			"curl", "-v", "-m", "3", "--fail", "localhost:4001")
```

Comment:
that's important. Sometimes we received 503 which is valid curl response therefore err was nil. `--fail` changes this behaviour so we will see retry for 503